chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -675,10 +693,10 @@ impl<'a, S: TransactionVerifierStorageRef, U: UtxosView> TransactionVerifier<'a,
         // TODO: test spending block rewards from chains outside the mainchain
         if let Some(inputs) = reward_transactable.inputs() {
             let tx_index_fetcher =
-                |tx_id: &OutPointSourceId| self.storage_ref.get_mainchain_tx_index(tx_id);
+                |tx_id: &OutPointSourceId| self.storage.get_mainchain_tx_index(tx_id);
```

Comment:
Just a quick coding style question - why do you create a fetcher here, but on line 664 you inline instead?

Reply:
That's just a result of me trying out different approaches to refactoring this. Will make it more consistent, thanks.